pkg/envoy/sds/response_test.go

```diff
@@ -337,43 +308,28 @@ func TestGetSDSSecrets(t *testing.T) {
 				tc.prepare(&d)
 			}
 
-			s := &sdsImpl{
-				serviceIdentity: tc.serviceIdentity,
-				certManager:     fakeCertManager,
-
-				// these points to the dynamic mocks which gets updated for each test
-				meshCatalog: d.mockCatalog,
-				TrustDomain: "cluster.local",
-			}
+			builder := newSdsBuilder()
+			builder.SetRequestedCerts(tc.requestedCerts)
 
 			proxy := envoy.NewProxy(envoy.KindSidecar, uuid.New(), identity.New("sa-1", "ns-1"), nil, 1)
+			builder.SetProxy(proxy)
+
+			builder.SetProxyCert(cert)
+			builder.SetTrustDomain("cluster.local")
+
+			serviceIdentitiesForServices := getServiceIdentitiesForServices(tc.requestedCerts, d.mockCatalog)
+			builder.SetServiceIdentitiesForService(serviceIdentitiesForServices)
 
 			// test the function
-			sdsSecrets := s.getSDSSecrets(cert, tc.requestedCerts, proxy)
+			// sdsSecrets := s.getSDSSecrets(cert, tc.requestedCerts, proxy)
+			sdsSecrets := builder.Build()
 			assert.Len(sdsSecrets, tc.expectedSecretCount)
 
 			if tc.expectedSecretCount <= 0 {
 				// nothing to validate further
 				return
 			}
 
-			sdsSecret := sdsSecrets[0]
-
-			// verify different cert types
-			switch tc.sdsCertType {
-			// Verify SAN for inbound and outbound MTLS certs
-			case secrets.RootCertTypeForMTLSInbound, secrets.RootCertTypeForMTLSOutbound:
-				// Check SANs
-				actualSANs := subjectAltNamesToStr(sdsSecret.GetValidationContext().GetMatchSubjectAltNames())
-				assert.ElementsMatch(actualSANs, tc.expectedSANs)
-
-				// Check trusted CA
-				assert.NotNil(sdsSecret.GetValidationContext().GetTrustedCa().GetInlineBytes())
-
-			case secrets.ServiceCertType:
-				assert.NotNil(sdsSecret.GetTlsCertificate().GetCertificateChain().GetInlineBytes())
-				assert.NotNil(sdsSecret.GetTlsCertificate().GetPrivateKey().GetInlineBytes())
-			}
```

Comment:
Why are these being removed? I don't see them being tested with the new changes.